component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,220 @@
+                    User user = new User();
+                    String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername
+                            (responseDTO.getAuthorizedUser());
+                    user.setUserName(UserCoreUtil.removeDomainFromName(tenantAwareUsername));
+                    user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(tenantAwareUsername));
+                    user.setTenantDomain(MultitenantUtils.getTenantDomain(responseDTO.getAuthorizedUser()));
+                    authenticationContext.setUser(user);
+                }
+
+                authenticationContext.addParameter(DPoPConstants.CONSUMER_KEY, consumerKey);
+                authenticationContext.addParameter(Constants.OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
+                authenticationContext.addParameter(Constants.OAUTH2_VALIDATE_SCOPE,
+                        AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                String serviceProvider = null;
+
+                try {
+                    serviceProvider =
+                            OAuth2Util.getServiceProvider(consumerKey).getApplicationName();
+                } catch (IdentityOAuth2Exception e) {
+                    String error = String.format("Error occurred while getting the Service Provider" +
+                            " by Consumer key: %s", consumerKey);
+                    log.error(error, e);
+                }
+
+                String serviceProviderTenantDomain = null;
+                try {
+                    serviceProviderTenantDomain =
+                            OAuth2Util.getTenantDomainOfOauthApp(consumerKey);
+                } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
+
+                    String error = String.format("Error occurred while getting the OAuth App" +
+                            " tenantDomain by Consumer key: %s", consumerKey);
+                    log.error(error, e);
+                }
+
+                if (serviceProvider != null) {
+                    authenticationContext.addParameter(DPoPConstants.SERVICE_PROVIDER, serviceProvider);
+                    if (serviceProviderTenantDomain != null) {
+                        authenticationContext.addParameter(DPoPConstants.SERVICE_PROVIDER_TENANT_DOMAIN,
+                                serviceProviderTenantDomain);
+                    }
+
+                    MDC.put(DPoPConstants.SERVICE_PROVIDER, serviceProvider);
+                    // Set OAuth service provider details to be consumed by the provisioning framework.
+                    setProvisioningServiceProviderThreadLocal(clientApplicationDTO.getConsumerKey(),
+                            serviceProviderTenantDomain);
+                }
+            }
+        }
+        return authenticationResult;
+    }
+
+    private AuthenticationResult getAuthenticationResult(AuthenticationResult authenticationResult,
```

Comment:
All public methods should be listed first, then the private methods come.

Reply:
Fixed with [a4c65c3](https://github.com/kalanika/identity-oauth-addons/commit/a4c65c37b8b7eb980f5cb1b9569a27044541faed).